patient_data_source.py

```diff
@@ -0,0 +1,214 @@
+from abc import ABC, abstractmethod
+import names
+import warnings
+from datetime import datetime
+from enum import Enum
+from collections.abc import Iterable
+from fhirclient.models.patient import Patient as FHIR_Patient
+from fhirclient.models.observation import Observation as FHIR_Observation
+from fhirclient.models.fhirdate import FHIRDate
+
+class Patient(ABC):
+	class Gender(Enum):
+		male = 0
+		female = 1
+		other = 2
+		unknown = 3
+
+	def get_gender(self) -> Gender:
+		"""Returns the patient's gender. Default implementation return Gender.unkown"""		
+		return Gender.unknown
+
+	def get_identifier_value(self) -> str:
+		"""Returns a id value for external use."""
+		return None
+
+	@abstractmethod
+	def get_dob(self) -> datetime:
+		"""Returns the patient's date of birth."""
+		return datetime.min
+
+	def get_identifier_system(self) -> str:
+		"""Returns the namespace for unique identifier values."""
```

Comment:
It's not very clear what "namespace" means. I know the FHIR documentation sometimes uses this obtuse language, but can afford to be a little more human readable here.

How about:
"Returns a description of the way to interpret the custom identifiers returned by `get_identifier_value`"
or somethign like that